rpmlint/checks/SharedLibraryPolicyCheck.py

```diff
@@ -0,0 +1,174 @@
+                    if not lib_part:
+                        continue
+                    if lib_part[-1].isdigit() and not lib_part.endswith('lib64'):
+                        libs.remove(lib)
+                        break
+
+        # Check for non-versioned libs in a std lib package
+        if std_lib_package:
+            for lib in libs.copy():
+                if (not (lib[-1].isdigit() or
+                         self.re_soname_strongly_versioned.search(lib))):
+                    self.output.add_info('W', pkg, 'shlib-unversioned-lib', lib)
+
+        if not pkg.name.startswith('lib') or pkg.name.endswith('-lang'):
+            return
+
+        # Verify no non-lib stuff is in the package
+        dirs = set()
+        for f in pkg.files:
+            if os.path.isdir(pkg.dirName() + f):
+                dirs.add(f)
+
+        # Verify shared lib policy package doesn't have hard dependency on non-lib packages
+        if std_lib_package:
+            for dep in pkg.requires:
+                if dep[0].startswith('rpmlib(') or dep[0].startswith('config('):
+                    continue
+                if (dep[1] & (rpm.RPMSENSE_GREATER | rpm.RPMSENSE_EQUAL)) == rpm.RPMSENSE_EQUAL:
+                    self.output.add_info('W', pkg, 'shlib-fixed-dependency', formatRequire(dep[0], dep[1], dep[2]))
+
+        # Verify non-lib stuff does not add dependencies
+        if libs:
+            for dep in pkg_requires:
+                if '.so.' in dep and dep not in libs and dep not in libs_needed:
+                    self.output.add_info('E', pkg, 'shlib-policy-excessive-dependency', dep)
+
+        # Check if the files/folders are unversioned in the library package.
+        # In general you can't co-install the soname packages if they all provide some datafiles
+        # or configuration files.
+        # When testing one of examples is libsemanage1:
+        #   /etc/selinux
+        #   /etc/selinux/semanage.conf
+        #   /usr/lib64/libsemanage.so.1
+        # The above would be fine if the semanage.conf would be update-alternatived, or suffixed
+        # but if someone introduces libsemanage2 they can't be installed both at once.
+
+        # Check for non-versioned directories beyond sysdirs in package
+        sysdirs = ('/lib/', '/lib64/', '/usr/lib/', '/usr/lib64/',
+                   '/usr/share/', '/usr/share/licenses/',
+                   '/usr/share/doc/')
+        cdirs = set()
+        for sysdir in sysdirs:
+            done = set()
+            for directory in dirs:
+                if directory.startswith(sysdir):
+                    ssdir = directory[len(sysdir):].split('/')[0]
+                    if not ssdir[-1].isdigit():
+                        cdirs.add(sysdir + ssdir)
+                    done.add(directory)
+            dirs = dirs.difference(done)
+        for directory in cdirs:
+            self.output.add_info('E', pkg, 'shlib-policy-nonversioned-dir', directory)
```

Comment:
this is prone to many mistakes, thats why i didnt finish it up. it should act as per above comment, but now does not.

Reply:
I debug the section and it works (somehow). What's wrong with that exactly?

Reply:
just search for this error across the whole tumbleweed. It is very commonly wrong not picking up, or picking too many things.

Reply:
As by the example in the comment above.

Reply:
I see, do you suggest to remove it for now?

Reply:
Ok, I removed the patch and added a `FIXME`. Hope it's fine now.